OIIInspector/ImageManager.py

```diff
@@ -0,0 +1,151 @@
+import logging
+from retry import retry
+import subprocess
+import time
+import shlex
+from contextlib import suppress
+from OIIInspector.Exceptions import AddressAlreadyInUse, OIIInspectorError
+from OIIInspector.utils import run_cmd
+
+log = logging.getLogger(__name__)
+
+
+class ImageManager:
+    _grpc_start_port = 50051
+    _grpc_max_port_tries = 100
+    _grpc_max_tries = 5
+    _grpc_init_wait_time = 1000
+    _base_container_name = "OIIInspector_running_image"
+
+    def __init__(self):
+        self._container_platform = self._get_container_platform()
+        self._image_address = None
+        self._port = self._grpc_start_port
+        self._rpc_proc = None
+
+    def start_image(self, image_address):
+        self._image_address = image_address
```

Comment:
Maybe you can move this to the constructor?

Reply:
You mean move all the start_image method to __init__ right? I thought that current solution is better as there is clear when the image is started. Is there some performance improvement? Or is it better practice in python? (This view is probably from my Java experiences)

Reply:
now start_image is called from __enter__ method, Is that a good solution?

Reply:
Ah, sorry, I just meant setting the image address.